### PR TITLE
Initial support for the SparkFun LoRa Thing Plus - expLoRaBLE

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

This adds support for the [SparkFun LoRa Thing Plus - expLoRaBLE](https://www.sparkfun.com/products/17506).

This board is similar to the existing [SparkFun RedBoard Artemis Nano](https://github.com/tock/tock/tree/master/boards/redboard_artemis_nano), except has a slightly different form factor, different pin outs and includes LoRa support via a SPI attached Semtech SX1262.

The existing code has been restructured to allow as much code sharing between the different Ambiq boards as possible.

### Testing Strategy

Running the Tock unit tests on the board. I was able to monitor the broken out I2C and SPI buses and see traffic with a logic analyser.

I haven't tested the SX1262 yet, but the board does have SPI support and the pins should be configured and enabled. 

### TODO or Help Wanted

Nothing todo for this PR, next step is to work on LoRa support

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

**Comment by bradjc** [comment]
LGTM, but I think we should hold off on adding a new board for a few days due to merge conflicts with pending PRs (specifically appid, but component changes as well).

File: boards/apollo3/lora_things_plus/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+/// This function is called from the arch crate after some very basic RISC-V
+/// setup and RAM initialization.
+#[no_mangle]
+pub unsafe fn main() {
+    #[cfg(test)]
+    test_main();
+
+    #[cfg(not(test))]
+    {
+        let (board_kernel, esp32_c3_board, chip, _peripherals) = setup();
```
Comment (twilfredo) [inline boards/apollo3/lora_things_plus/src/main.rs:454]:
```suggestion
        let (board_kernel, sf_lora_thing_plus_board, chip, _peripherals) = setup();
```
Should probably change this here too https://github.com/tock/tock/blob/09600e9deab75e1b3faa7eec2be26cbebae2d2cf/boards/redboard_artemis_nano/src/main.rs#L433

File: boards/apollo3/lora_things_plus/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+
+        let main_loop_cap = create_capability!(capabilities::MainLoopCapability);
+
+        board_kernel.kernel_loop(
+            esp32_c3_board,
```
Comment (twilfredo) [inline boards/apollo3/lora_things_plus/src/main.rs:459]:
```suggestion
            sf_lora_thing_plus_board,
```

File: boards/apollo3/lora_things_plus/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+    }
+}
+
+#[cfg(test)]
+use kernel::platform::watchdog::WatchDog;
+
+#[cfg(test)]
+fn test_runner(tests: &[&dyn Fn()]) {
+    unsafe {
+        let (board_kernel, esp32_c3_board, _chip, peripherals) = setup();
```
Comment (twilfredo) [inline boards/apollo3/lora_things_plus/src/main.rs:473]:
```suggestion
        let (board_kernel, sf_lora_thing_plus_board, _chip, peripherals) = setup();
```

File: boards/apollo3/lora_things_plus/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+
+        BOARD = Some(board_kernel);
+        PLATFORM = Some(&esp32_c3_board);
```
Comment (twilfredo) [inline boards/apollo3/lora_things_plus/src/main.rs:476]:
```suggestion
        PLATFORM = Some(&sf_lora_thing_plus_board);
```

**Review by bradjc — APPROVED** [review]

**Review by twilfredo — DISMISSED** [review]

**Review by twilfredo — APPROVED** [review]

**Review by hudson-ayers — APPROVED** [review]
bors r+

**Pull request merged**